exasol/bucketfs/_buckets.py

```diff
@@ -223,6 +232,84 @@ def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
             yield from response.iter_content(chunk_size=chunk_size)
 
 
+class SaaSBucket:
+
+    def __init__(self, url: str, account_id: str, database_id: str, pat: str) -> None:
+        self._url = url
+        self._account_id = account_id
+        self.database_id = database_id
+        self._pat = pat
+
+    def name(self) -> str:
+        # TODO: Find out the name of the bucket in SaaS
+        return 'default'
+
+    def files(self) -> Iterable[str]:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def delete(self, path: str) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def upload(self, path: str, data: ByteString | BinaryIO) -> None:
+        """To be provided"""
+        raise NotImplementedError()
+
+    def download(self, path: str, chunk_size: int = 8192) -> Iterable[ByteString]:
+        """To be provided"""
+        raise NotImplementedError()
+
+
+class MountedBucket:
+    """
+    Implementation of the Bucket interface backed by a normal file system in read-only mode.
+    The targeted use case is the read-only access to the BucketFS files from a UDF.
+
+    Arguments:
+        service_name:
+            Name of the BucketFS service (not a service url). Defaults to 'bfsdefault'.
+        bucket_name:
+            Name of the bucket. Defaults to 'default'.
+
+    Raises MountedBucketFsError on attempt to call a BucketLike method that would require
+    a write permission.
+    """
+
+    def __init__(self,
+                 service_name: str = 'bfsdefault',
+                 bucket_name: str = 'default'):
+        self._name = bucket_name
+        self.root = Path('buckets') / service_name / bucket_name
```

Comment:
With this we get at least two problems.

1. the path will be relative to the working directory.
2. This is now fixed to be only used in UDFs, we cannot use it for mocking the BucketFS when we test with the udf-mock-python, but maybe we want to have a dedicated implementation for that

Reply:
Re:1 In this case yes, it will be relative to the working directory. That's where I have been asking for opinions on whether we should or should not include the service_name / bucket_name in the path.
Re:2 No, the class is still used as a base for BucketFake in unit testing.